libdnf/transaction/CompsEnvironmentItem.cpp

```diff
@@ -187,8 +187,10 @@ CompsEnvironmentItem::getTransactionItemsByPattern(SQLite3Ptr conn, const std::s
 
     // HACK: create a private connection to avoid undefined behavior
     // after forking process in Anaconda
-    SQLite3 privateConn(conn->getPath());
-    SQLite3::Query query(privateConn, sql);
+    if (conn->getPath() != ":memory:") {
```

Comment:
Is there any report that it can happen?

Reply:
Under a regular user that doesn't have the permissions to access the history database (either is not readable or doesn't exist and they don't have write permissions). Realistically it's a rare corner case but it can happen.